### PR TITLE
Feature/cu j191b2 module navigation

### PR DESCRIPTION
**Pull request by magalenyo**
[](url)Module Recast and Detour integrated to project.

NavMeshes can be baked with the new **PanelNavigation**. In this Panel, a NavMesh build configuration can be set. 

Once the **NavMesh** is built and saved, it **must be set as a Resource in the Scene Configuration panel.**

In order to have GameObjects navigate the scene, a new **ComponentAgent** can be added to GameObject. ComponentAgent has a speed and acceleration (which will make the agent have an inertia).
ComponentAgent has the following public functions: **SetMaxSpeed**, **SetMaxAcceleration**, **SetMoveTarget**(target, usePathFinding=true). For PlayerController, when moving the player, SetMoveTarget should have usePathFinding as false.

**IMPORTANT!** There's a gizmos toggle to check the NavMesh generated, but it will replace the view of the existing meshes due to conflicts with Immediate mode rendering.

Zombunny.scene is a test scene to check the functionality of the NavMesh. ZombunnyController is an example of usage.

Gizmos:
70896139544783e75215c7c02db96e59

Test:
1b16d8599f17d8d8ac1f10001c9dc2f7

**Note 1**: In order to avoid the "bounce back" effect, a high acceleration must be set to the agent (p.eg: 999). Max speed will be the regular speed.
**Take into account that changing acceleration may not fix the bounce effect if speed is too high**. If it's the case, acceleration will have to be set to a higher ammount. This will be fixed in future evolutionaries.

**Note 2**: Once the NavMesh is built and saved, the navMesh will be saved with the build configuration applied, but the configuration itself **WON'T BE SAVED** (yet). Please, take note of the build parameters before exiting the application.

### CODE REVIEW

File: Project/Source/Resources/ResourceNavMesh.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "Application.h"
+#include "Modules/ModuleFiles.h"
+#include "Modules/ModuleNavigation.h"
+#include "Modules/ModuleScene.h"
+#include "Components/ComponentAgent.h"
+#include "Scene.h"
+#include "Utils/Logging.h"
+#include "Utils/Buffer.h"
+#include "Utils/MSTimer.h"
```
Comment (JaimeBea) [inline Project/Source/Resources/ResourceNavMesh.cpp:11]: This file is missing Leaks.h

**Review by JaimeBea — DISMISSED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by jcarlos0305 — DISMISSED** [review]

**Comment by FrancPS** [comment]
Check the warnings

**Review by FrancPS — DISMISSED** [review]

**Review by FrancPS — APPROVED** [review]

**Review by JaimeBea — DISMISSED** [review]

**Review by Jordi-Gil — APPROVED** [review]
Memory leaks when bake multiple times the same mesh

image

**Comment by JaimeBea** [comment]
> Memory leaks when bake multiple times the same mesh
> 
> image

I can't seem to reproduce the leaks. When does this happen?

**Comment by Jordi-Gil** [comment]
Open the engine --> Play --> Stop --> Close the engine. It seems is not related to this PR.

**Comment by Jordi-Gil** [comment]
We have those memory leaks in develop (without any script), probably its something related to the temporary scene generation.

**Comment by FrancPS** [comment]
They come from Physics. We were trying to fix them and we found that or they come from the Bullet library or they are not leaks".
Comment the stepSimulation(dt) in ModulePhysiscs.cpp -- PreUpdate(). Thats the function causing them

**Review by FrancPS — DISMISSED** [review]

**Review by JaimeBea — APPROVED** [review]

**Review by FrancPS — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Pull request merged**